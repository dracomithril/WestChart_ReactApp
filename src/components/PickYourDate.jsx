/**
 * Created by Gryzli on 12.02.2017.
 */
import React from "react";
import PropTypes from "prop-types";
import {Checkbox, OverlayTrigger, Popover, Tooltip, Button} from "react-bootstrap";
import DatePicker from "react-datepicker";
export default class PickYourDate extends React.Component {
    /*istanbul ignore next*/
    componentWillUnmount() {
        console.log('component ChartPresenter unmounted');
    }

    /*istanbul ignore next*/
    componentDidMount() {
        console.log('component ChartPresenter did mount');
    }

    render() {
        const {store} = this.context;
        const {enable_until, start_date, show_last} = store.getState();
        const footer = <div className="datePiker">
            <Checkbox checked={enable_until} name="enable_until"
                      onChange={(e) => store.dispatch({
                          type: 'TOGGLE_ENABLE_UNTIL',
                          checked: e.target.checked
                      })}>{'Use date: '}
                <DatePicker
                    selected={start_date}
                    dateFormat="DD/MM/YYYY"
                    onChange={date => store.dispatch({type: 'UPDATE_START_TIME', date})}
                    disabled={!enable_until}/>
            </Checkbox>
        </div>;
        return (<div id="pickYourDate">
                <OverlayTrigger trigger={["hover", "focus"]} overlay={<Tooltip id="go_back">
                    {'How far in time you will travel'}
                </Tooltip>}>
                    <span style={{paddingRight:10}}>{'go back '}
                        <input className="num_days"
                               type="number"
                               name="show_last"
                               min={0} max={62}
                               value={show_last}
                               step={1}
                               onChange={(e) => {
                                   store.dispatch({type: 'UPDATE_SHOW_LAST', days: Number(e.target.value)})
                               }}/>{' days'}
                    </span>
                </OverlayTrigger>
                <OverlayTrigger trigger={"click"} placement="bottom" overlay={<Popover id="more_options_chose_date">
                    {footer}
                </Popover>}>
                    <Button className="fa fa-angle-double-down" bsSize="small"/>
                </OverlayTrigger>
            </div>);
    }
}
PickYourDate.contextTypes = {
    store: PropTypes.object.isRequired
};
PickYourDate.propTypes = {};