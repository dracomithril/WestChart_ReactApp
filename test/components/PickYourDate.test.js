/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from "react";
import PropTypes from "prop-types";
import PickYourDate from "../../src/components/PickYourDate";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import configureMockStore from "redux-mock-store";
jest.mock('./../../src/utils');
const initial_state = require('../data/initial_state.json');
const mockStore = configureMockStore([]);
describe('<PickYourDate/>', () => {
    afterAll(() => {
        jest.unmock('./../../src/utils');
    });
    it('renders without crashing', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <PickYourDate />, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('renders without crashing', () => {
        const store = mockStore(initial_state);
        const utils = require('./../../src/utils');
        const wrapper = shallow(
            <PickYourDate />, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        let button = wrapper.find('#updateChartB');

        button.simulate('click');
        let actions = store.getActions();
        expect(actions.length).toBe(3);
        expect(utils.getChartFromServer.mock.calls.length).toBe(1);
    });
});
