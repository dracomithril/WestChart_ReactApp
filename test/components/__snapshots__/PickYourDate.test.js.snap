// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PickYourDate/> renders without crashing 1`] = `
<Accordion>
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}
    footer={
      <small
        id="updateDate"
      >
         Last update: Invalid Date
      </small>
    }
    header="Pick your date"
  >
    <label>
      How far in time you will travel 
      <input
        className="num_days"
        max={62}
        min={0}
        name="show_last"
        onChange={[Function]}
        step={1}
        type="number"
        value={31}
      />
       days
    </label>
    <Checkbox
      bsClass="checkbox"
      disabled={false}
      inline={false}
      name="enable_until"
      onChange={[Function]}
    >
      Use date: 
      <t
        dateFormat="DD/MM/YYYY"
        dateFormatCalendar="MMMM YYYY"
        disabled={true}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        monthsShown={1}
        onBlur={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        popoverAttachment="top left"
        popoverTargetAttachment="bottom left"
        popoverTargetOffset="10px 0"
        tetherConstraints={
          Array [
            Object {
              "attachment": "together",
              "to": "window",
            },
          ]
        }
        utcOffset={120}
        withPortal={false}
      />
    </Checkbox>
    <div>
      <Label
        bsClass="label"
        bsStyle="success"
      >
        since: Invalid Date
      </Label>
      <br />
      <Label
        bsClass="label"
        bsStyle="danger"
        id="d_until"
      >
        until: 3/11/2017, 2:46:00 PM
      </Label>
    </div>
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="primary"
        disabled={false}
        id="updateChartB"
        onClick={[Function]}
      >
        Update
      </Button>
    </div>
  </Panel>
</Accordion>
`;
